File: delightreading-server/src/main/java/com/delightreading/user/hibernate/ExperienceListUserType.java
package com.delightreading.user.hibernate;

import com.delightreading.common.JsonBasedType;
import com.delightreading.user.Experience;
import com.fasterxml.jackson.core.type.TypeReference;

import java.util.List;

public class ExperienceListUserType extends JsonBasedType {
    public ExperienceListUserType() {
        super(new TypeReference<List<Experience>>(){});
    }

    @Override
//    public Class<List<Experience>> returnedClass() {
//        return Experience.class;
//    }
    public Class<List> returnedClass() {
        return List.class;
    }
}

File: delightreading-backend/src/main/kotlin/com/delightreading/user/hibernate/ExperienceListUserType.kt
package com.delightreading.user.hibernate

import com.delightreading.common.JsonBasedType
import com.delightreading.user.model.Experience
import com.fasterxml.jackson.core.type.TypeReference

class ExperienceListUserType : JsonBasedType(object : TypeReference<List<Experience>>() {
}) {

    override fun returnedClass(): Class<List<*>> {
        return List::class.java
    }
}